controllers/apidefinition_controller.go

```diff
@@ -180,6 +156,50 @@ func (r *ApiDefinitionReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	return ctrl.Result{RequeueAfter: queueA}, err
 }
 
+func (r *ApiDefinitionReconciler) checkSecretAndUpload(
+	ctx context.Context,
+	certName string,
+	namespacedName types.NamespacedName,
+	log logr.Logger,
+	env *environmet.Env,
+) (string, error, bool) {
+	secret := v1.Secret{}
+
+	err := r.Get(ctx, types.NamespacedName{Name: certName, Namespace: namespacedName.Namespace}, &secret)
+	if err != nil {
+		log.Error(err, "requeueing because secret not found")
+
+		return "", err, true
+	}
+
+	pemCrtBytes, ok := secret.Data["tls.crt"]
+	if !ok {
+		log.Error(err, "requeueing because cert not found in secret")
+
+		return "", err, true
+	}
+
+	pemKeyBytes, ok := secret.Data["tls.key"]
+	if !ok {
+		log.Error(err, "requeueing because key not found in secret")
```

Comment:
We need to log an error, but `err` is nil, because `pemKeyBytes, ok := secret.Data["tls.key"]` doesn't set err

Reply:
I have updated the PR with the required changes.